North-Seattle-College/ad440-winter2021-tuesday-repo
# Converted Connections to Live

### What:
This makes the React UI pull live data, for better or for worse. That is, it polls the results from 8 functions (and there are only 4).

[Users](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/users?)
[Specific User](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/users/1?)
[That user's tasks](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/users/1/tasks?)
[That task](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/users/{userId}/tasks/1?)
[Artillery Tests - NOT IMPLEMENTED](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/artillery?)
[Specific Artillery Test - NOT IMPLEMENTED](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/artillery/1?)
[Serverless Tests - NOT IMPLEMENTED](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/serverless?)
[Specific Serverless Test - NOT IMPLEMENTED](https://nsc-func-dev-usw2-tuesday.azurewebsites.net/api/serverless/1?)


All usage of dummy data has been deleted (hence the giant red bar).

### How to run and test using React
1. Install [nodejs](https://www.nodejs.org). Typically, use the LTS version unless you are sure of what you are doing.
2. Install an editor like [vscode](https://code.visualstudio.com/).
3. Install [git](https://git-scm.com/downloads). This allows you to clone this current respository.
4. Navigate to a folder that you would like the project file in.
5. In the top left corner (on windows), replace the folder location with ```cmd``` and press enter. On Mac, options-clicking on the folder you wish to open will allow you to open it in terminal (should you have this enabled in settings).
6. ```git clone https://github.com/North-Seattle-College/ad440-winter2021-tuesday-repo.git```
7. Open VS Code.
8. Open the folder in VSCode by pressing CTRL-O (Windows) and navigating to the new folder.
9. CTRL-SHIFT-` (Windows) to open the terminal.
10. ```cd ui``` should navigate you to the UI folder.
11. ```npm install``` to install the project for local development.
12. When this is complete, ```npm start``` to run the project locally.
13. You can navigate through the website using the navlinks on the left. As we currently have no backend supporting the serverless and artillery containers, these pages will be blank. However, it is connecting.


### OR:
Remote Website, should this be merged and moved to the server: 
https://nscstrdevusw2tuecommon.z5.web.core.windows.net/
